lib/MusicBrainz/Server/Controller/Artist.pm

```diff
@@ -362,15 +363,25 @@ Shows all works of an artist.
 sub works : Chained('load')
 {
     my ($self, $c) = @_;
+
+    my $artist = $c->stash->{artist};
+    my %filter = %{ $self->process_filter($c, sub {
+        return create_artist_works_form($c, $artist->id);
+    }) };
+    my $has_filter = %filter ? 1 : 0;
```

Comment:
Since `process_filter` already returns a hash ref and you need one to pass to `find_by_artist`, it seems simpler to use it as returned
```suggestion
    my $artist = $c->stash->{artist};
    my $filter = $self->process_filter($c, sub {
        return create_artist_works_form($c, $artist->id);
    });
    my $has_filter = %$filter ? 1 : 0;
```